src/Test/TestDumbMath.java

```diff
@@ -0,0 +1,98 @@
+    GameScenario scn = new GameScenario();
+    Army[] armies = new Army[mapInfo.getNumCos()];
+    armies[0] = new Army(scn, CommandingOfficers.AWBW.BM.Olaf.getInfo().create(scn.rules));
+    armies[1] = new Army(scn, CommandingOfficers.AWBW.BM.Colin.getInfo().create(scn.rules));
+    armies[2] = new Army(scn, CommandingOfficers.AW2.BM.Colin.getInfo().create(scn.rules));
+    armies[3] = new Army(scn, CommandingOfficers.AW3.BM.Colin.getInfo().create(scn.rules));
+
+    olaf    = armies[0].cos[0];
+    colinBW = armies[1].cos[0];
+    colin2  = armies[2].cos[0];
+    colin3  = armies[3].cos[0];
+
+    map  = new MapMaster(armies, mapInfo);
+    game = new GameInstance(armies, map);
+
+    turn(game);
+
+    boolean testPassed = true;
+
+    UnitContext tankOlaf = new UnitContext(addUnit(map, olaf, UnitModel.ASSAULT, 4, 7)); // On road
+
+    UnitContext tankBW   = new UnitContext(colinBW, colinBW.getUnitModel(UnitModel.ASSAULT, false));
+    UnitContext tank2    = new UnitContext(colin2, colin2.getUnitModel(UnitModel.ASSAULT, false));
+    UnitContext tank3    = new UnitContext(colin3, colin3.getUnitModel(UnitModel.ASSAULT, false));
+
+    BattleParams shotBW = new CombatContext(game, map, tankBW, tankOlaf, 1, CalcType.NO_LUCK).applyModifiers().getAttack();
+    BattleParams shot2  = new CombatContext(game, map, tank2 , tankOlaf, 1, CalcType.NO_LUCK).applyModifiers().getAttack();
+    BattleParams shot3  = new CombatContext(game, map, tank3 , tankOlaf, 1, CalcType.NO_LUCK).applyModifiers().getAttack();
+
+    shotBW.attackPower += 10;
+    shotBW.attackPower += CommandingOfficers.AWBW.BM.Colin.calcSuperBoost(241700);
+    int hitBW = shotBW.calculateDamage();
+    testPassed &= validate(hitBW == 453,     "    ColinBW PoM is wrong");
+
+    shot3.attackPower += 10;
+    shot3.attackPower += CommandingOfficers.AW3.BM.Colin.calcSuperBoost(241700);
+    int hit3 = shot3.calculateDamage();
+    testPassed &= validate(hit3 == 497,     "    Colin3 PoM is wrong");
+
+    shot2.attackerDamageMultiplier = CommandingOfficers.AW2.BM.Colin.calcSuperMult(240900);
+    int hit2 = shot2.calculateDamage();
+    testPassed &= validate(hit2 == 442,     "    Colin2 PoM is wrong");
+    shot2.attackerDamageMultiplier = CommandingOfficers.AW2.BM.Colin.calcSuperMult(431660);
+    hit2 = shot2.calculateDamage();
+    testPassed &= validate(hit2 == 753,     "    Colin2 PoM is wrong");
+    shot2.attackerDamageMultiplier = CommandingOfficers.AW2.BM.Colin.calcSuperMult(110000);
+    hit2 = shot2.calculateDamage();
+    testPassed &= validate(hit2 == 228,     "    Colin2 PoM is wrong");
+    shot2.attackerHealth = 40;
+    hit2 = shot2.calculateDamage();
+    testPassed &= validate(hit2 ==  91,     "    Colin2 PoM is wrong");
+    shot2.attackerDamageMultiplier = CommandingOfficers.AW2.BM.Colin.calcSuperMult(640660);
+    shot2.attackerHealth = 60;
+    hit2 = shot2.calculateDamage();
+    testPassed &= validate(hit2 == 657,     "    Colin2 PoM is wrong");
+
```

Comment:
How'd you pick the funds amounts for these tests? And why use different ones for each test? How am I supposed to know which Colin is best Colin when numbers not same?

Reply:
I played some matches (badly) on cart, and tested numbers with savestates.
They aren't the same numbers because math is hard.

Best Colin is unambiguously DS Colin since he charges Gold Rush faster, and it doesn't really matter whether you have +1000 attack or +1100 attack during PoM... but his PoM is also strongest:
AW2 PoM is weakest.
AWBW and DS Colin PoM are close, but \*3/1000 < \*1/300, respectively.

Reply:
Ah, makes sense. As long as we answer the important question here.